### ISSUE TITLE
Data resource github_repositories only queries public repos

### ISSUE THREAD

**Issue by Geethree** [terraform-providers/terraform-provider-github#409]
### Terraform Version
```
$ terraform providers -v
Terraform v0.12.24
+ provider.aws v2.55.0
+ provider.github v2.5.0
```

### Affected Resource(s)
- [github_repositories](https://www.terraform.io/docs/providers/github/d/repositories.html)

### Terraform Configuration Files
```hcl
data "github_repositories" "all_repos" {
  query = "org:foo"
}

output repos {
  value = data.github_repositories.all_repos.names
}
```

### Expected Behavior
`org:foo` is an organization with only private repositories. I expected the data source to utilize the token and org provided to find all repositories that exist in the org.

### Actual Behavior
No repositories were found

### Important Factoids
We have another legacy organization with a mixture of private and public repos. When calling the query on that org (using a token with full access), I get a list of only the public repositories.

**Comment by Geethree** [comment]
Basically, I'd like to query all repos in the org and set each team's permissions with the `github_team_repository` resource. 

- Admins are admins on all
- Developers have maintain on all
- Contractors have write on all

Kind of thing.

**Comment by Geethree** [comment]
After poking around more to the differences between the `github_repository` and `github_repositories` data sources, it was determined that they utilize the [same method to search for repositories via the github client](https://github.com/terraform-providers/terraform-provider-github/blob/a372e8adfa7d81bf1978298cfaaceaec5fb7db15/vendor/github.com/google/go-github/v29/github/search.go#L73) - which makes sense!

So when I went to call one of these private repositories via the `github_repository` data resource, I ended up getting a
```
Error: GET https://api.github.com/repos/foo/bar: 404 Not Found []
```

And sure enough, the token did not have private repo access, only org access. When adjusting the permissions of the token, the `github_repositories` data source returns all of the private repositories in org foo.

Not sure as the best way to document this functionality, but, perhaps, one could modify the data source to take in another variable..

- `repo_visibility` with the options of _public_, _private_ with the default set to public? This would potentially call out this fact to users of this data resource.

It appears that the [Github Search API includes a way to filter based on public/private](https://help.github.com/en/github/searching-for-information-on-github/searching-for-repositories#search-by-public-or-private-repository) via:
```
is:public|private
```

Anyways, closing for now.
